starter/src/main/java/com/vivy/springdoc/SwaggerUiPropertiesListener.java

```diff
@@ -0,0 +1,19 @@
+package com.vivy.springdoc;
+
+import org.springframework.boot.context.event.ApplicationEnvironmentPreparedEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+
+import java.util.Properties;
+
+public class SwaggerUiPropertiesListener implements ApplicationListener<ApplicationEnvironmentPreparedEvent> {
+
+    @Override
+    public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event) {
```

Comment:
not the best way of adding props to a context. You could add a props contributor bean and register that in already existing configuration

Reply:
you can use `@PropertySource("classpath:custom.properties")` on the config class and just add your file with a props file providing what you need